lib/nerves_network/if_supervisor.ex

```diff
@@ -37,13 +38,18 @@ defmodule Nerves.Network.IFSupervisor do
     end
   end
 
+  def scan(ifname) when is_atom(ifname), do: scan(to_string(ifname))
+
   def scan(ifname) do
-     pidname = pname(ifname)
-     if Process.whereis(pidname) do
-       GenServer.call(pidname, :scan, 30_000)
-     else
-       {:error, :not_started}
-     end
+    with pid when is_pid(pid) <- Process.whereis(pname(ifname)),
```

Comment:
I'm wondering if at this point, would it be needed to check `ifname in Nerves.NetworkInterface.interfaces()`?

Reply:
I think that the request should be routed through the WiFiManager since it will know the state of the supplicant. Likewise, the non-WiFi managers can error out the scan call.

Reply:
ah i just found the _real_ problem in `WiFiManager`. Got another commit coming